### PR TITLE
Add handling for training keras models in mnist_blackbox tutorial

### PR DESCRIPTION
**Pull request by aspratyush**
The patch in #398, also discussed in #400, adds `model.get_params()` to `model_train`. However, since `get_params()` is a tensorflow specific API, it causes a regression when using Keras-based model. This PR checks the model type, uses a conditional `model_train` implementation, and also sets `model.trainable=False` when using Keras-based models.

### CODE REVIEW

**Comment by npapernot** [comment]
Thanks for the PR. Unfortunately, we decided a few months ago to remove all Keras dependencies from the CleverHans module and tutorials to the exception of two files: `cleverhans/utils_keras.py` and `cleverhans_tutorials/mnist_tutorial_keras_tf.py`. This is primarily because supporting Keras created a lot of overhead so I would rather move a blackbox tutorial using Keras to the `examples/` folder where we have looser constraints for merging code. 

@goodfeli  any opinion?

**Comment by goodfeli** [comment]
Yes, I agree. I don't want to provide any ongoing support for more files using Keras.

@aspratyush , please let us know whether you'd prefer to move this to `examples/` (where there is no obligation for ongoing support) or whether you'd prefer to close the PR.

In general, it's a good idea to write to cleverhans-dev@googlegroups.com before starting work on a feature to make sure it will be approved.

**Comment by npapernot** [comment]
@aspratyush Do you want to move this file to `examples/` or should I close this PR? Thanks!

**Comment by aspratyush** [comment]
@npapernot : Have moved the keras-based black-box model file to `examples/`

While with the recent update to `cleverhans.utils_tf.train` that makes `var_list` an optional parameter resolves this specific regression, usage of `cleverhans.loss.LossCrossEntropy` requires that the defined model be an instance of `cleverhans.model.Model`. Hence, a Keras-based black-box model would need to be wrapped in a `cleverhans.utils_keras.KerasModelWrapper` to successfully run this example.

This updated PR addresses handling such an implementation.

**Comment by npapernot** [comment]
Thanks!
